pyfr/plugins/turbulence.py

```diff
@@ -0,0 +1,255 @@
+                            tmp = actbuff['buff'][:,geid][actbuff['buff'][:,geid]['te'] > tcurr]        
+                            shft = actbuff['nvmx']-len(tmp)   
+                            if shft:
+                                newb = np.zeros(shft, self.buffdtype)
+                                temp = self.vortbuff[['tinit', 'state']][sstream['vid'][:shft]]
+                                pad = shft-temp.shape[0]
+                                newb[['tinit', 'state']] = np.pad(temp, (0,pad), 'constant')
+                                newb[['ts', 'te']] = np.pad(sstream[['ts', 'te']][:shft], (0,pad), 'constant')
+                                self.actbuffs[abid]['buff'][:,geid] = np.concatenate((tmp,newb))
+                                self.actbuffs[abid]['sstream'][geid] = sstream[shft:]
+                            else:
+                                tstemp = sstream['ts'][0]
+                                if tcurr >= tstemp:
+                                    print(f'Active vortex in stream cannot fit on buffer.')
+                                
+                            if self.actbuffs[abid]['sstream'][geid]['vid'].any() and (self.actbuffs[abid]['buff'][-1,geid]['ts'] < trcl):
+                                trcl = self.actbuffs[abid]['buff'][-1,geid]['ts']
+
+                    self.actbuffs[abid]['trcl'] = trcl
+                    self.actbuffs[abid]['tinit'].set(actbuff['buff']['tinit'][:, np.newaxis, :])
+                    self.actbuffs[abid]['state'].set(actbuff['buff']['state'][:, np.newaxis, :])
+            
+            proptnext = min(etype['trcl'] for etype in self.actbuffs)
+            if proptnext > self.tnext:
+                self.tnext = proptnext
+            else:
+                print('Not advancing.')
+
+    def vortbuff(self):
+        pcg32rng = pcg32rxs_m_xs(self.seed)
+        vid = 0
+        temp = []
+        xtemp = []
+        
+        tinits = []
+        
+        while vid < self.nvorts:
+            tinits.append(self.tstart + (self.xmax-self.xmin)*pcg32rng.random()/self.ubar)
+            vid += 1
+
+        while True:     
+            for vid, tinit in enumerate(tinits):
+                state = pcg32rng.getstate()
+                yinit = self.ymin + (self.ymax-self.ymin)*pcg32rng.random()
+                zinit = self.zmin + (self.zmax-self.zmin)*pcg32rng.random()
+                if tinit+((self.xmax-self.xmin)/self.ubar) >= self.tbegin and tinit <= self.tend:
+                    xtemp.append(((yinit,zinit),tinit,state))
+                tinits[vid] += (self.xmax-self.xmin)/self.ubar
+            if all(tinit > self.tend for tinit in tinits):
+                break
+    
+        return np.asarray(xtemp, self.vortdtype)
+
+    def actbuffs(self, intg):
+        actbuffs = []
+        for etype, eles in intg.system.ele_map.items(): 
+            neles = eles.neles
+            pts = eles.ploc_at_np('upts')
+            pts = np.moveaxis(pts, 1, 0)
```

Comment:
Should be able to combine with the previous line and convert to a swapaxis method call.